src/particle.cpp

```diff
@@ -391,6 +406,46 @@ void Particle::event_death()
   }
 }
 
+void Particle::pht_collision_energy()
+{
+  // determine index of cell in pulse_height_cells
+  auto it = std::find(model::pulse_height_cells.begin(),
+    model::pulse_height_cells.end(), coord(n_coord() - 1).cell);
+  int index = std::distance(model::pulse_height_cells.begin(), it);
+  // Adds the energy particles lose in a collision to the pulse-height at the
+  // cell index
+  pht_storage()[index] += E_last() - E();
+
+  // If the energy of the particle is below the cutoff, it will not be sampled
+  // so its energy is added to the pulse-height in the cell
+  if (E() < settings::energy_cutoff[1]) {
+    pht_storage()[index] += E();
+  }
+}
+
+void Particle::pht_secondary_particles()
+{
+  // Removes the energy of secondary produced particles from the pulse-height
+
+  // determine the birth cell of the particle
+  if (coord(n_coord() - 1).cell == C_NONE) {
+    if (!exhaustive_find_cell(*this)) {
+      mark_as_lost(
+        "Could not find the cell containing particle " + std::to_string(id()));
+      return;
+    }
+
+    // Set birth cell attribute
+    if (cell_born() == C_NONE)
+      cell_born() = coord(n_coord() - 1).cell;
```

Comment:
We had a discussion on this already in the old version of the pht:
https://github.com/openmc-dev/openmc/pull/1881/files#r726887201

Reply:
I have simulated a simple example (happy to share) and currently the `cell_born` attribute is __not__ corresponding for the original source particle. If a secondary particle is created in a cell the `cell_born` attribute is set to this cell for the secondary particle.

Maybe we should add a small comment into the docs?
Instead of 
_Bins tally events based on which cell the particle was born in._
maybe the following solves some confusion:
_Bins tally events based on which cell the particle was born in. For secondary particles this attribute is set to the cell in which they were produced._

Reply:
This is how the `cell_born` attribute is currently handled in my understanding:
1) particles start with a `None` entry.
2) in the `transport_history_based_single_particle` function we are calling in the first step of the loop always the `p.event_calculate_xs()` method. In this method, the `cell_born` attribute is set.

In the case of the pulse-height tally we have to remove the energy of secondary particles that were created, cause their energy may not remain in the cell. So whenever a particle is created we remove the energy directly from the pulse_height value. 
Since this attribute is set from a None to the cell the particle was born in only a bit earlier, this will not change the behaviour of the `CellBornFilter`. 
I see that this is not really elegant. But changing it would require a method to distinguish whether a particle is a primary or a secondary particle. Therefore I would recommend to leave it as it is.